hazelcast/src/main/java/com/hazelcast/map/impl/proxy/MapProxySupport.java

```diff
@@ -156,7 +157,14 @@ protected MapProxySupport(final String name, final MapService service, NodeEngin
         partitionStrategy = mapServiceContext.getMapContainer(name).getPartitioningStrategy();
         localMapStats = mapServiceContext.getLocalMapStatsProvider().getLocalMapStatsImpl(name);
         this.partitionService = getNodeEngine().getPartitionService();
-        lockSupport = new LockProxySupport(new DefaultObjectNamespace(MapService.SERVICE_NAME, name));
+
+        LockService lockService = nodeEngine.getSharedService(LockService.SERVICE_NAME);
+        if (lockService != null) {
+            lockSupport = new LockProxySupport(new DefaultObjectNamespace(MapService.SERVICE_NAME, name),
+                    lockService.getMaxLeaseTimeInMillis());
+        } else {
+            lockSupport = null;
```

Comment:
Afais `IMap#lock` may throw NPE after this change because of `lockSupport  = null` case.  That case is a new thing introduced with this PR? intentional?